### PR TITLE
Update the readme for v5

### PR DESCRIPTION
**Pull request by notlee**
This aims to document `o-forms` with migration guide.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,139 +20,188 @@ Each form field is made up of at least 3 elements:
...
+```html
+<div class="o-forms">
+	<label for="o-forms-demo-optional" class="o-forms__label o-forms__label--optional">Optional  input example</label>
+	<input type="text" id="o-forms-demo-optional" placeholder="placeholder" class="o-forms__text" />
+</div>
+```
+
+#### Wide fields
+
+The class `.o-forms--wide` can be used to show form fields without width restrictions. This is demonstarted below with a text input but can apply to any `.o-forms` field.
```
Comment (gvonkoss) [inline README.md]: typo demonstarted -> demonstrated

File: README.md [hunk 2]
```diff
@@ -20,139 +20,188 @@ Each form field is made up of at least 3 elements:
...
+		<option value="option4">Option 4</option>
+	</select>
+</div>
+```
+
+[Select boxes examples](https://www.ft.com/__origami/service/build/v2/demos/o-forms/select-boxes)
+
+#### Checkbox and radio controls
+
+Unlike other inputs, multiple `checkbox` and `radio` inputs should be wrapped in  `.o-forms__group` to ensure correct verticle spacing.
```
Comment (gvonkoss) [inline README.md]: typo verticle -> vertical

File: README.md [hunk 3]
```diff
@@ -20,139 +20,188 @@ Each form field is made up of at least 3 elements:
...
 </fieldset>
 ```
 
-Add a wrapper and data attribute `data-o-form-toggle` to checkboxes to make them look like toggles
+[Radio control examples](https://www.ft.com/__origami/service/build/v2/demos/o-forms/radios)
+[Checkbox examples](https://www.ft.com/__origami/service/build/v2/demos/o-forms/checkboxes)
+
+#### Suffixes
+
+Suffixes are used to append content to an input, i.e. a button. Add a wrapper `.o-forms__affix-wrapper` to contain the input and the `.o-forms__suffix` itself.
```
Comment (gvonkoss) [inline README.md]: do you mean add the suffix class to the input itself?

File: README.md [hunk 4]
```diff
@@ -20,139 +20,188 @@ Each form field is made up of at least 3 elements:
...
-[Radio control examples](https://www.ft.com/__origami/service/build/v2/demos/o-forms/radios)
-[Checkbox examples](https://www.ft.com/__origami/service/build/v2/demos/o-forms/checkboxes)
-[Toggle Checkbox examples](https://www.ft.com/__origami/service/build/v2/demos/o-forms/toggle)
+[Suffixes examples](https://www.ft.com/__origami/service/build/v2/demos/o-forms/suffix)
 
-#### Optional Labels
+#### Toggles (checkbox toggles)
 
-To indicated an input is optional apply the class `.o-forms__label--optional` to its label.
+Wrap checkboxes in `.o-forms__toggle` and add the data attribute `data-o-form-toggle` to produce a toggle interface.
```
Comment (gvonkoss) [inline README.md]: what element is the attribute being added to?

File: README.md [hunk 5]
```diff
@@ -161,107 +210,76 @@ Example HTML:
...
 ### Sass
 
 #### Silent mode
 
 In silent mode `o-forms` provides mixins for each set of form fields as well as some mixins to output basic form styles in one large chunk.
 
-#### oForms mixin
-
-The `oForms` mixin will output styles for the `o-forms` block, as well as all basic input styles supported by `o-forms` (`text`, `select`, `textarea`, `checkboxes` and `radio` buttons) as well as `labels`, valid and error states and, error text and helper text.
+The `oForms` mixin will output all features of `o-forms`. Turn off silent mode to output all `o-forms` features using this mixin automatically.
```
Comment (gvonkoss) [inline README.md]: should it be `oForms()`?

Comment (notlee) [reply README.md]: Debatable.  Its only parameter is optional so you could `@include oForms` or  `@include oForms()`.

**Comment by notlee** [comment]
Thanks @gvonkoss! I've corrected those issues and snuck in the migration guide, if you wouldn't mind reviewing that also? :)

File: README.md [hunk 6]
```diff
@@ -349,14 +349,48 @@ There are a number of inconsistencies in how browsers handle form events, valida
...
+- The `oFormsCheckboxToggleSize` mixin has been removed due to lack of use.
+
+### Wrappers and Messages
+- Wrappers have been renamed to sections. Their class names have also been updated to conform to the BEM naming convention (as optional containers their name should not contain `__` as they are not elements of a block).
+	- `.o-forms__wrapper` becomes `.o-forms-section`.
+	- `.o-forms__wrapper--highlight` becomes `.o-forms-section--highlight`.
+	- `.o-forms__wrapper--error` which becomes `.o-forms-section--error`.
+	- The `oFormsMessage` mixin now only outputs minimal message styles, uses should be replaced with `oFormsSectionFeature`.
+- Messages are now child elements of a section and must not be used independently.
+	- Wrap messages within a form seciton `.o-forms-section` if they are not already. They should be the first child of the section.
```
Comment (gvonkoss) [inline README.md]: typo seciton -> section

**Pull request merged**